src/test/java/fi/otavanopisto/kuntaapi/test/AbstractMocker.java

```diff
@@ -99,82 +93,64 @@ public void mockGetString(String path, String type, String content, Map<String,
     } catch (JsonProcessingException e) {
       logger.log(Level.SEVERE, "Failed to serialize mock JSON object", e);
       fail(e.getMessage());
     }
   }
   
   /**
-   * Reads JSON file as organization object
-   * 
-   * @param file path to JSON file
-   */    
-  public Organization readOrganizationFromJSONFile(String file) {
-    return readJSONFile(file, Organization.class);
-  }
-  
-  /**
-   * Reads JSON file as organization object
-   * 
-   * @param file path to JSON file
-   */    
-  public Service readServiceFromJSONFile(String file) {
-    return readJSONFile(file, Service.class);
-  }
-  
-  /**
-   * Reads JSON file as electronic service channel object
+   * Mocks XML response for GET request on path
    * 
-   * @param file path to JSON file
-   */    
-  public ElectronicChannel readElectronicChannelFromJSONFile(String file) {
-    return readJSONFile(file, ElectronicChannel.class);
+   * @param path path
+   * @param object JSON object
+   */
+  public void mockGetXML(String path, Object object, Map<String, String> queryParams) {
+    try {
+      XmlMapper xmlMapper = new XmlMapper();
+      xmlMapper.registerModule(new JavaTimeModule());
+      stringMocks.add(new StringGetMock(path, "text/xml", xmlMapper.writeValueAsString(object), queryParams));
+    } catch (JsonProcessingException e) {
+      logger.log(Level.SEVERE, "Failed to serialize mock JSON object", e);
+      fail(e.getMessage());
+    }
   }
   
-  /**
-   * Reads JSON file as phone service channel object
-   * 
-   * @param file path to JSON file
-   */    
-  public PhoneChannel readPhoneChannelFromJSONFile(String file) {
-    return readJSONFile(file, PhoneChannel.class);
-  }
-
-  /**
-   * Reads JSON file as printable form service channel object
-   * 
-   * @param file path to JSON file
-   */    
-  public PrintableFormChannel readPrintableFormChannelFromJSONFile(String file) {
-    return readJSONFile(file, PrintableFormChannel.class);
-  }
-
-  /**
-   * Reads JSON file as printable form service channel object
-   * 
-   * @param file path to JSON file
-   */    
-  public ServiceLocationChannel readServiceLocationChannelFromJSONFile(String file) {
-    return readJSONFile(file, ServiceLocationChannel.class);
-  }
-
-  /**
-   * Reads JSON file as printable form service channel object
-   * 
-   * @param file path to JSON file
-   */    
-  public WebPageChannel readWebPageChannelFromJSONFile(String file) {
-    return readJSONFile(file, WebPageChannel.class);
+  protected <T> T readXMLFile(String file, Class <T> type){
+    XmlMapper xmlMapper = new XmlMapper();
+    xmlMapper.registerModule(new JavaTimeModule());
+    
+    try (InputStream stream = getClass().getClassLoader().getResourceAsStream(file)) {
+      return xmlMapper.readValue(stream, type);
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Failed to read mock file", e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Failed to read mock file" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)